src/Service/Client/ServiceClient.php

```diff
@@ -74,11 +74,27 @@ public function request(
         array $scopes = []
     ): ResponseInterface {
         try {
-            $options = array_merge_recursive($options, [
-                'headers' => ['Authorization' => sprintf('Bearer %s', $this->getAccessToken($registration, $scopes))]
-            ]);
+            $options = array_merge_recursive(
+                $options,
+                [
+                    'headers' => [
+                        'Authorization' => sprintf('Bearer %s', $this->getAccessToken($registration, $scopes))
+                    ]
+                ]
+            );
+
+            $response = $this->client->request($method, $uri, $options);
+
+            if ($response->getStatusCode() === 401) {
+                $options['headers']['Authorization'] = sprintf(
+                    'Bearer %s',
+                    $this->getAccessToken($registration, $scopes, true)
+                );
```

Comment:
the ServiceClient allows to inject your own guzzle client instance, which potential already applied middlewares could collide this the auth header appending.

to ensure / enforce this header to be added, instead of complexifying this set up via external middleware, I act directly on request options to be sure I'll have this header as expected.